src/main/cpp/src/zorder.cu

```diff
@@ -27,6 +27,118 @@
 #include <thrust/iterator/constant_iterator.h>
 #include <thrust/iterator/counting_iterator.h>
 
+namespace {
+
+// pretends to be an array of uint32_t, but really only stores
+// the data in a long with a set number of bits allocated for
+// each item (num_bits_per_entry)
+struct long_backed_array {
+  long_backed_array() = delete;
+  ~long_backed_array() = default;
+  long_backed_array(long_backed_array const&) = default;  ///< Copy constructor
+  long_backed_array(long_backed_array&&) = default;  ///< Move constructor
```

Comment:
I believe that you don't have to declare these default methods. Only declare what you need.

Reply:
They are just to dilute your code. Except for the overloaded constructor (you provide one then the compiler will not provide the default constructor), other methods are implicitly generated the same as you write `default`. So explicitly writing them `... = default` is redundant.